### PR TITLE
Add Microsoft.CSharp tests for more access permutations.

### PR DESCRIPTION
**Pull request by JonHanna**
Some access cases not currently covered for fields and nested types.

### CODE REVIEW

**Comment by JonHanna** [comment]
Test OSX x64 Debug Build

**Comment by JonHanna** [comment]
Test OSX x64 Debug Build

File: src/Microsoft.CSharp/tests/AccessTests.cs [hunk 1]
```diff
@@ -40,6 +40,18 @@ private struct PrivateValueType
...
 
+            private protected struct PrivateProtectedValueType
+            {
+            }
+
+            internal struct InternalValueType
+            {
+            }
+
+            protected internal struct InternalProtectedValueType
```
Comment (stephentoub) [inline src/Microsoft.CSharp/tests/AccessTests.cs]: Nit: ProtectedInternalValueType?

**Comment by stephentoub** [comment]
Thanks

**Pull request merged**